### PR TITLE
Add delete snapshot controller UT

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <grant@portworx.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
* adds one delete test for common_controller

**Which issue(s) this PR fixes**:
Part of #191 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/common-controller/snapshot_delete_test.go [hunk 1]
```diff
@@ -316,6 +322,48 @@ func TestDeleteSync(t *testing.T) {
...
+					Status: &crdv1.VolumeSnapshotStatus{
+						CreationTime:                   nil,
+						ReadyToUse:                     &False,
+						Error:                          nil,
+						RestoreSize:                    nil,
+						BoundVolumeSnapshotContentName: &content31,
+					},
+				},
+			},
+			//expectedSnapshots: newSnapshotArray("snap3-1", "snapuid3-1", "claim3-1", "", validSecretClass, "content3-1", &False, nil, nil, nil, false, true, &timeNowMetav1),
```
[the remark this answers is not in the document]
Comment (xing-yang) [reply pkg/common-controller/snapshot_delete_test.go]: Also tests 1-4 and 1-5 are not needed and can be removed.

[the remark this answers is not in the document]
Comment (ggriffiths) [reply pkg/common-controller/snapshot_delete_test.go]: Addressed these

**Pull request merged**